### PR TITLE
Fix to allow comment poster to edit or delete his own comments

### LINKED ISSUE #11663: Allow users to edit their own comments, without needing full Write permission
- Gitea version (or commit ref): 1.11.3
- Git version: 2.24.1
- Operating system: Linux (Docker)
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes (provide example URL)
  - [ ] No
  - [ ] Not relevant
- Log gist:

## Description

I have some people with access to a repository, that can report issues, but only a developer can resolve the issues.

These people are unable to edit/delete their own comments (to fix typos, or add more information, or delete comment that was added into the wrong issue)

I would like for them to be able to edit/delete their own comments, without, needing to give them the full Write access.

Maybe it's similar to this other request https://discourse.gitea.io/t/how-to-allow-collaborators-to-edit-their-own-comments-in-1-5-x/581

### PR DESCRIPTION
**Pull request by a1012112796**
Now the ui only allow poster to update or delet comment, which is not reasonable and different with handle logic, this pr change it to allow poster of comment do it.

ref code:
https://github.com/go-gitea/gitea/blob/e8955173a9be1acaa9a3755c37b6059422acda20/routers/repo/issue.go#L1636

https://github.com/go-gitea/gitea/blob/e8955173a9be1acaa9a3755c37b6059422acda20/routers/repo/issue.go#L1681

fix #11663

### CODE REVIEW

**Review by lafriks — APPROVED** [review]

File: templates/repo/diff/comments.tmpl [hunk 1]
```diff
@@ -31,7 +31,7 @@
 				{{end}}
 			{{end}}
 			{{template "repo/issue/view_content/add_reaction" Dict "ctx" $ "ActionURL" (Printf "%s/comments/%d/reactions" $.root.RepoLink .ID) }}
-			{{template "repo/issue/view_content/context_menu" Dict "ctx" $.root "item" . "delete" true "diff" true }}
+			{{template "repo/issue/view_content/context_menu" Dict "ctx" $.root "item" . "delete" true "diff" true "IsCommentPoster" .IsCommentPoster}}
```
Comment (lafriks) [inline templates/repo/diff/comments.tmpl]:
```suggestion
			{{template "repo/issue/view_content/context_menu" Dict "ctx" $.root "item" . "delete" true "diff" true "IsCommentPoster" ($.root.IsUser .PosterID)}}
```

**Review by lafriks — CHANGES_REQUESTED** [review]
I think it should be better to add function to context.go:
```go
// IsUser returns true if current user is specified one
func (ctx *Context) IsUser(id int64) bool {
	return ctx.IsSigned && (ctx.User.ID == id)
}
```

**Comment by a1012112796** [comment]
@lafriks I think this style is simpler. Do you think so? Thanks.

**Review by lafriks — APPROVED** [review]

**Comment by lafriks** [comment]
What needs to be verified is if `SignedUserID` does not equals to ghost/deleted or imported user comment poster id

File: templates/repo/diff/comments.tmpl [hunk 2]
```diff
@@ -31,7 +31,7 @@
 				{{end}}
 			{{end}}
 			{{template "repo/issue/view_content/add_reaction" Dict "ctx" $ "ActionURL" (Printf "%s/comments/%d/reactions" $.root.RepoLink .ID) }}
-			{{template "repo/issue/view_content/context_menu" Dict "ctx" $.root "item" . "delete" true "diff" true }}
+			{{template "repo/issue/view_content/context_menu" Dict "ctx" $.root "item" . "delete" true "diff" true "IsCommentPoster" (eq $.root.SignedUserID .PosterID)}}
```
Comment (zeripath) [inline templates/repo/diff/comments.tmpl]:
```suggestion
			{{template "repo/issue/view_content/context_menu" Dict "ctx" $.root "item" . "delete" true "diff" true "IsCommentPoster" (and $.root.IsSigned (eq $.root.SignedUserID .PosterID))}}
```

File: templates/repo/issue/view_content/comments.tmpl [hunk 3]
```diff
@@ -39,7 +39,7 @@
 								</div>
 							{{end}}
 							{{template "repo/issue/view_content/add_reaction" Dict "ctx" $ "ActionURL" (Printf "%s/comments/%d/reactions" $.RepoLink .ID)}}
-							{{template "repo/issue/view_content/context_menu" Dict "ctx" $ "item" . "delete" true "diff" false }}
+							{{template "repo/issue/view_content/context_menu" Dict "ctx" $ "item" . "delete" true "diff" false "IsCommentPoster" (eq $.SignedUserID .PosterID)}}
```
Comment (zeripath) [inline templates/repo/issue/view_content/comments.tmpl]:
```suggestion
							{{template "repo/issue/view_content/context_menu" Dict "ctx" $ "item" . "delete" true "diff" false "IsCommentPoster" (and $.root.IsSigned (eq $.SignedUserID .PosterID))}}
```

**Comment by a1012112796** [comment]
Please review, Thanks :)

**Comment by 6543** [comment]
didnt this worked before too?!?

**Review by 6543 — APPROVED** [review]

**Comment by lafriks** [comment]
make lg-tm work

**Comment by lafriks** [comment]
make it work

**Comment by lafriks** [comment]
Please send backport

**Pull request merged**